src/sys/socket/sockopt.rs

```diff
@@ -432,7 +432,19 @@ sockopt_impl!(
     #[allow(missing_docs)]
     // Not documented by Linux!
     Ip6tOriginalDst, GetOnly, libc::SOL_IPV6, libc::IP6T_SO_ORIGINAL_DST, libc::sockaddr_in6);
-sockopt_impl!( 
+#[cfg(any(target_os = "linux"))]
+sockopt_impl!(
+    /// Specifies exact type of timestamping information collected by the kernel
+    /// [Further reading](https://www.kernel.org/doc/html/latest/networking/timestamping.html)
+    /// Takes a bitwise `or` of libc timestamping options:
+    /// - [SOF_TIMESTAMPING_TX_HARDWARE][::libc::SOF_TIMESTAMPING_TX_HARDWARE]:
+    /// - [SOF_TIMESTAMPING_TX_SOFTWARE][::libc::SOF_TIMESTAMPING_TX_SOFTWARE]:
+    /// - [SOF_TIMESTAMPING_RX_HARDWARE][::libc::SOF_TIMESTAMPING_RX_HARDWARE]:
+    /// - [SOF_TIMESTAMPING_RX_SOFTWARE][::libc::SOF_TIMESTAMPING_RX_SOFTWARE]:
+    /// - [SOF_TIMESTAMPING_RAW_HARDWARE][::libc::SOF_TIMESTAMPING_RAW_HARDWARE]:
+    /// - [SOF_TIMESTAMPING_SOFTWARE][::libc::SOF_TIMESTAMPING_SOFTWARE]
+    Timestamping, Both, libc::SOL_SOCKET, libc::SO_TIMESTAMPING, u32);
```

Comment:
Instead of taking a `u32` argument, you should use `libc_bitflags!` to define a custom type.